examples/Resources/Private/Singles/Variables.html

```diff
@@ -40,6 +40,12 @@
 	}
 }"/>
 
+<!-- Passing arguments to a sections/partials using f:render in tag content -->
+<f:render section="Tertiary">
+    <f:variable name="arg1">First argument</f:variable>
```

Comment:
I am not sure if we really want to overload this viewhelper with this additional magic. What if one wants to have some template code here, simplified using `f:variable` and then pass everything via `contentAs` as usual? Of course, one does not have to use all arguments passed to a section/partial but we should avoid passing around too much data.

I'd prefer a more explicit approach, e.g. with a new `f:argument` or `f:render.argument` viewhelper.

Reply:
This relates to the https://github.com/TYPO3/Fluid/issues/424 vision which calls for `f:argument` to be used for another purpose. I would personally like to keep the distinction that a "variable" is something you insert/assign, whereas an "argument" is something you declare/pass.

> What if one wants to have some template code here, simplified using `f:variable` and then pass everything via `contentAs` as usual?

Then the combined arguments received by the partial would also contain those "variables" you defined while simplifying your template. The pitfall would imho be that if you named such a variable the same as an argument then your nested variable overrules the argument.

Perhaps it is more suitable the other way around: arguments passed in the array take priority over nested variables?

In any case I would not make such a case into an exception or other error.